### PR TITLE
Feature: Added support for bulk renaming items

### LINKED ISSUE #8191: Feature: Support renaming multiple files at the same time
### What's the Problem?

Only one file can be renamed at the same time.

### Solution/Idea

1. Using the rename action when multiple items are selected should open a "Rename items" dialog
2. The rename dialog should consist of a simple title and text box for entering the new name
3. Items containing different extensions should get the new name
4. Items with the same extension should have a number attached to the end

### PR DESCRIPTION
**Pull request by marianalcar**
**Resolved / Related Issues**

Previously, only one file could be renamed at the same time. We implemented the functionality of renaming multiple at the same time giving Files more flexibility and increased functionalities.

- Closes #8191 

**Steps used to test these changes**

1. Opened Files
2. Went to each layout view
3. Created different files and folders
4. Selected them all and renamed them at the same time.

### CODE REVIEW

File: src/Files.App/Actions/FileSystem/RenameAction.cs [hunk 1]
```diff
@@ -33,15 +33,22 @@ public RenameAction()
 		}
 
 		public Task ExecuteAsync(object? parameter = null)
-		{
-			context.ShellPage?.SlimContentPage?.ItemManipulationModel.StartRenameItem();
+		{	
+			if (context.SelectedItems.Count == 1)
```
Comment (yaira2) [inline src/Files.App/Actions/FileSystem/RenameAction.cs:37]: Is this if statement needed? It looks like the else statement has the same logic.

File: src/Files.App/Actions/FileSystem/RenameAction.cs [hunk 2]
```diff
@@ -33,15 +33,22 @@ public RenameAction()
...
+			{
+				context.ShellPage?.SlimContentPage?.ItemManipulationModel.StartRenameItem();
```
Comment (yaira2) [inline src/Files.App/Actions/FileSystem/RenameAction.cs:39]: What are your thoughts on moving the logic to within `StartRenameItem`?

**Comment by 0x5bfa** [comment]
Could you provide a screen recording?

**Comment by yaira2** [comment]
> Using the rename action when multiple items are selected should open a "Rename items" dialog

I compiled this branch, but it seems to be missing the dialog when bulk renaming.

**Comment by marianalcar** [comment]
> Could you provide a screen recording?

https://github.com/files-community/Files/assets/114589228/01b8d4f8-7f8b-4724-b706-bc0067c49113

File: src/Files.App/Actions/FileSystem/RenameAction.cs (hunk 2 above)
Comment (marianalcar) [reply src/Files.App/Actions/FileSystem/RenameAction.cs]: We commited that change now. Thank you for the feedback

**Comment by yaira2** [comment]
@marianalcar the behavior seems to closely mimic File Explorer, while it technically works, we were hoping to add more advanced options in the future and therefore this feature requires a content dialog.

File: src/Files.App/Actions/FileSystem/RenameAction.cs (hunk 2 above)
Comment (yaira2) [reply src/Files.App/Actions/FileSystem/RenameAction.cs]: Thanks!

**Comment by yaira2** [comment]
Sorry about the confusion! For now, the rename dialog should consist of a simple title and text box for entering the new name. In the future, we may want to add more options.

**Comment by yaira2** [comment]
@marianalcar are you still planning to work on this?

**Comment by yaira2** [comment]
@marianalcar I've made the requested changes in #16228. Can you take a look and comment if you have any feedback?

**Pull request closed without merging**